signup.go

```diff
@@ -382,11 +383,15 @@ func (sc *SignupService) register(ctx context.Context, su Signup) (Signup, error
 	su.ShortLink = shortLink
 
 	// Run each task in a go routine for concurrent execution
-	g, ctx := errgroup.WithContext(ctx)
-	for _, task := range sc.tasks {
+	// Creating a new context because the errgroup will cancel the context when
+	// Wait() is returns.
```

Comment:
was this automatically put on a new line? it took me a second to realize it was part of the comment above

```suggestion
	// Wait() is returned.
```